fmskill/comparison.py

```diff
@@ -977,15 +981,37 @@ def scatter(
             skill_df = self.skill(
                 metrics=["bias", "rmse", "urmse", "mae", "cc", "si", "r2"], df=df
             )  # df is filtered to matching subset
+            stats_with_units=["bias", "rmse", "urmse", "mae"]
+            
             lines = []
-
+            if units==None:
+                #Check for units
+                try:
+                    #Extract first letter of most common dfs0 units in SI;
+                    #either (m)eter, (s)econd, (deg)gree, or (m/s). Rest of units can
+                    #always be assigned by user with `units=str` keyword
+                    units_=self._itemInfos[0].unit.display_name
+                    if units_=='meter' or units_=='second':
+                        units_=units_[0]
+                    elif units_=='degree':
+                        units_='°'
+                    elif units_=='meter per sec':
+                        units_='m/s'
+                except:
+                    #Dimensionless
+                    units_=''
             max_str_len = skill_df.df.columns.str.len().max()
 
             for col in skill_df.df.columns:
                 if col == "model":
                     continue
-                lines.append(
-                    f"{col.ljust(max_str_len)} {np.round(skill_df.df[col].values[0],3)}"
+                if col in stats_with_units:
+                    lines.append(
+                    f"{(col.ljust(max_str_len)).upper()} {np.round(skill_df.df[col].values[0],3)} {units_}"
+                )
+                else:
+                    lines.append(
+                    f"{(col.ljust(max_str_len)).upper()} {np.round(skill_df.df[col].values[0],3)}"
```

Comment:
This line is almost identical to the one three lines above. Change it to make it clear what the difference between the cases is.

Reply:
I rewrote that part to make it clear in new commit